### ISSUE TITLE
Object (abstract) property wrongly seen as TypeVar or generic type in isinstance

### ISSUE THREAD

**Issue by Fatsie** [microsoft/pylance-release#1294]
## Environment data

- Language Server version: 2021.5.1 (pyright df8b6b3c)
- OS and version: Ubuntu 20.04
- Python version (& distribution if applicable, e.g. Anaconda): Python 3.7.9 64-bit (conda)

## Code Snippet

```python
from abc import abstractmethod
from typing import Any, Union, Generic, TypeVar, Type, Tuple

T = TypeVar("T")

class A(Generic[T]):
    @property
    @abstractmethod
    def _elem_type_(self) -> Union[Type[T], Tuple[Type[T], ...]]:
        raise NotImplementedError
    
    def check_type(self, var: Any) -> bool:
        return isinstance(var, self._elem_type_)
    
    def execute(self, var: Union[T, Tuple[T]]) -> None:
        if isinstance(var, self._elem_type_):
            print("single")
        else:
            print("iterable")
```

## Expected behaviour

Not give a Pylance error
## Actual behaviour

The use of `isinstance()` in `check_type()` method is OK; the use of `isinstance()` in `execute()` gives the following Pylance typing error:

```
{
	"resource": "/home/verhaegs/eda/test/test_generic.py",
	"owner": "_generated_diagnostic_collection_name_#2",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"external": "https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules",
			"path": "/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md",
			"scheme": "https",
			"authority": "github.com",
			"fragment": "diagnostic-severity-rules"
		}
	},
	"severity": 8,
	"message": "Second argument to \"isinstance\" must be a class or tuple of classes\n  TypeVar or generic type with type arguments not allowed",
	"source": "Pylance",
	"startLineNumber": 16,
	"startColumn": 28,
	"endLineNumber": 16,
	"endColumn": 44
}
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.5.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202153-19-may-2021
